R.NET/Devices/CharacterDeviceAdapter.cs

```diff
@@ -10,8 +10,18 @@ namespace RDotNet.Devices
 {
     internal class CharacterDeviceAdapter : IDisposable
     {
-        private readonly ICharacterDevice device;
+        /// <summary>
+        /// When R calls the character device (unamanged R calling managed code),
+        /// it sometimes calls the method with 'this == null' when writing/reading
+        /// from console (this seems to happen on Mono and may be a bug).
+        ///
+        /// The (somewhat incorrect) workaround is to keep the last device in a static
+        /// field and use it when 'this == null' (the check is done in 'this.Device').
+        /// This workarounds: http://rdotnet.codeplex.com/workitem/154
+        /// </summary>
+        private static ICharacterDevice lastDevice;

```

Comment:
It seems like a reasonable, if odd, workaround for the mono issue. I'm not sure what the correct path is, but I'm concerned a class static is rife for unintended caching side-effects. I wonder if we could constrain this to mono only?